app/models/post.rb

```diff
@@ -0,0 +1,5 @@
+class Post < ApplicationRecord
+    def to_param
+        slug
+    end
```

Comment:
by default, rails sets up `post_path(@post)` to use the `id` attribute of post. So I needed to override `to_param` in order to use `post_path(@post)` for `slug` instead of `id`